.gitignore

```diff
@@ -70,9 +70,11 @@ zulip.kdev4
 *.kate-swp
 *.sublime-project
 *.sublime-workspace
-.vscode/
 *.DS_Store
-# .cache/ is generated by Visual Studio Code's test runner
+# VS Code - generally ignore files, except for extension recs for the project
+/.vscode/**/*
+!/.vscode/extensions.json
```

Comment:
I think it's probably better to just keep this part as is, given that the `.react` example suggests one might want it; I think we initially added it for a reason, too.

Reply:
Like I said, I've never noticed noise files in that directory. That includes that I've never seen them in a PR, or heard about them from a contributor.

The .react subdirectory, I added to our gitignore in the first month or so I was working on the app:
zulip/zulip-mobile@ab6069d511c56a4d776d91507fa901925fd8db1a
apparently because it was in the setup instructions for an extension for React Native:
https://github.com/Microsoft/vscode-react-native#getting-started
Apparently the extension puts debugging files there. That's not an extension that is likely to be used in the zulip/zulip repo in any case.

One can certainly put project-specific config there. If so, it's generally best to share that too. Here's what we have in zulip-mobile:
```
$ cat .vscode/settings.json 
// Workspace Settings for the zulip-mobile repo.
{
    // Disable TS-centric validation.  We'll rely on ESLint for general validation.
    // See https://github.com/Microsoft/vscode/issues/15171
    "javascript.validate.enable": false,

    // Makes IntelliSense work with Flow.  See
    // https://github.com/Microsoft/vscode-react-native/blob/master/doc/intellisense.md .
    "flow.useNPMPackagedFlow": true,

    // Format on save (with prettier-eslint).
    "[javascript]": {
        "editor.formatOnSave": true,
    },
}
```